libs/azure-iot-middleware-freertos/source/include/azure_iot_adu_client.h

```diff
@@ -111,6 +111,36 @@
 #define azureiotaduUPDATE_NAME_SIZE                 24
 #define azureiotaduUPDATE_VERSION_SIZE              10
 
+/* ADU.200702.R: root production key from 7/2/2020  */
+static const unsigned char AzureIoTADURootKeyId[] = "ADU.200702.R";
+static const unsigned char AzureIoTADURootKeyN[] = { 0x00, 0xd5, 0x42, 0x2e, 0xaf, 0x11, 0x54, 0xa3, 0x50, 0x65, 0x87, 0xa2, 0x4d, 0x5b, 0xba,
```

Comment:
Is this OK declaring a (large) static in a .h like this?  I worry specifically that if multiple .c files #include and reference this variable, then there'll be a separate copy of each of these variables *per .C|.obj file.

In Visual Studio land, there's a mechanism of [`__declspec(selectany)`](https://docs.microsoft.com/en-us/cpp/cpp/selectany?view=msvc-170) to tell the linker to be smarter about this and just put this in one compilation unit.   I vaguely recall a discussion with our Azure SDK friends about this a while ago but I can't recall how other compilers - gcc/clang/some of the embedded specific ones - will do right thing here.  Maybe a discussion with Azure SDK unless you know?

Reply:
Yes that seems cleanest to solve this since there's no universal declspec for what MSVC is doing.